### ISSUE TITLE
ACM fails with cluster registered to a fleet in another project. 

### ISSUE THREAD

**Issue by cathchu** [terraform-google-modules/terraform-google-kubernetes-engine#1630]
### TL;DR

ACM fails with cluster registered to a fleet in another project. 

### Expected behavior

Setting `project_id` to the GCP project_id used to reach cluster (fleet project) would successfully install ConfigSync. 

### Observed behavior

Below, the ACM submodule is used to configure Config Sync and Policy Controller on an existing cluster, referenced as a data resource (data.google_container_cluster.gke.name):

```
module "acm" {

  source                    = "terraform-google-modules/kubernetes-engine/google//modules/acm"
  project_id                = "{CLUSTER_PROJECT_ID}" #var.gcp_project_id 
  cluster_name              = data.google_container_cluster.gke.name
  cluster_membership_id     = module.hub.cluster_membership_id
  enable_fleet_registration = false
  enable_fleet_feature      = false
  location                  = data.google_container_cluster.gke.location
  sync_repo                 = "{REPO}"
  sync_branch               = "main"
  policy_dir                = "config-root/all-clusters"
  enable_config_sync        = true
  enable_policy_controller  = true
  source_format             = "unstructured"
  create_ssh_key            = false 
  secret_type               = "token"
  metrics_gcp_sa_name       = "acm-${data.google_container_cluster.gke.name}"
  create_metrics_gcp_sa     = true

}
```

Fleet registration was successful, however, when installing ConfigSync, Terraform fails:

```
│ Error: Error creating FeatureMembership: googleapi: Error 404: Resource 'projects/{CLUSTER_PROJECT_ID}/locations/global/features/configmanagement' was not found
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.ResourceInfo",
│     "resourceName": "projects/{CLUSTER_PROJECT_ID}locations/global/features/configmanagement"
│   }
│ ]
│
│   with module.acm.google_gke_hub_feature_membership.main,
│   on .terraform/modules/acm/modules/acm/feature.tf line 26, in resource "google_gke_hub_feature_membership" "main":
│   26: resource "google_gke_hub_feature_membership" "main" {
```

We attempted to change the module so that project_id was set to the fleet_project_id, but then it failed in the terraform plan phase as the cluster_id is set to the fleet_project_id” 

```
Plan: 3 to add, 0 to change, 3 to destroy.
╷
│ Error: projects/{FLEET_PROJECT_ID}/locations/us-east4/clusters/gke-dev-dynatrace-01-us-east4 not found
│
│   with module.acm.module.registration.data.google_container_cluster.primary,
│   on .terraform/modules/acm/modules/fleet-membership/[main.tf](http://main.tf/) line 24, in data "google_container_cluster" "primary":
│   24: data "google_container_cluster" "primary" {
│
╵
```

### Terraform Configuration

```hcl
See above.
```

### Terraform Version

```sh
v1.4.4
```

### Additional information

Using the `gke_hub_feature_membership` resource did work with the following configuration:

```
resource "google_gke_hub_feature_membership" "feature_member" {
  provider   = google-beta
  location   = "global"
  feature    = "configmanagement"
  membership = google_gke_hub_membership.membership.membership_id
  project    = var.fleet_project_id
  configmanagement {
    version = "1.14.3"
    config_sync {
      source_format = "unstructured"
      git {
        sync_repo   = "{REPO}"
        sync_branch = "main"
        policy_dir  = "config-root/all-clusters"
        secret_type = "token"
      }
    }
    policy_controller {
      enabled                    = true
      template_library_installed = true
      referential_rules_enabled  = true
    }
  }
google_gke_hub_feature.configmanagement_acm_feature
]
}
```

**Comment by bharathkkb** [comment]
It is odd that the data source is trying to read `projects/{FLEET_PROJECT_ID}/locations/us-east4/clusters/gke-dev-dynatrace-01-us-east4 not found`.  What is the value provided via `cluster_name` variable? 

In the ACM module, that is plumbed to [hub registration](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/5abb6ba5b7a445c46b1b30efa5a7945996b39de4/modules/acm/main.tf#L22-L25) where it [reads the data source](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/fleet-membership/main.tf#L24-L27). FWIW, we could also make that data source read conditional for just creation of membership but I don't think that is the issue here.
